modules/docs/docgen/spec/modelParser.spec.ts

```diff
@@ -0,0 +1,162 @@
+    expect(symbols).toHaveProperty('1.type.modelProperties.0.name', 'getId');
+    expect(symbols).toHaveProperty('1.type.modelProperties.0.type.kind', 'function');
+    expect(symbols).toHaveProperty('1.type.modelProperties.0.type.parameters', []);
+    expect(symbols).toHaveProperty('1.type.modelProperties.0.type.returnType.kind', 'primitive');
+    expect(symbols).toHaveProperty('1.type.modelProperties.0.type.returnType.value', 'boolean');
+
+    // modelHook config
+    expect(symbols).toHaveProperty('2.name', 'IdModelConfig');
+    expect(symbols).toHaveProperty('2.type.kind', 'object');
+    expect(symbols).toHaveProperty('2.type.properties.0.name', 'a');
+    expect(symbols).toHaveProperty('2.type.properties.0.required', false);
+    expect(symbols).toHaveProperty('2.type.properties.0.defaultValue.kind', 'string');
+    expect(symbols).toHaveProperty('2.type.properties.0.defaultValue.value', '');
+    expect(symbols).toHaveProperty('2.type.properties.0.type.kind', 'primitive');
+    expect(symbols).toHaveProperty('2.type.properties.0.type.value', 'string');
+    expect(symbols).toHaveProperty('2.type.properties.1.name', 'b');
+    expect(symbols).toHaveProperty('2.type.properties.1.required', true);
+    expect(symbols).toHaveProperty('2.type.properties.1.type.kind', 'primitive');
+    expect(symbols).toHaveProperty('2.type.properties.1.type.value', 'string');
+    // callback
+    expect(symbols).toHaveProperty('2.type.properties.2.name', 'onOpen');
+    expect(symbols).toHaveProperty('2.type.properties.2.kind', 'property');
+    expect(symbols).toHaveProperty('2.type.properties.2.type.kind', 'function');
+    expect(symbols).toHaveProperty('2.type.properties.2.type.parameters.0.kind', 'parameter');
+    expect(symbols).toHaveProperty('2.type.properties.2.type.parameters.0.name', 'state');
+    expect(symbols).toHaveProperty('2.type.properties.2.type.parameters.0.type.kind', 'symbol');
+    expect(symbols).toHaveProperty(
+      '2.type.properties.2.type.parameters.0.type.name',
+      'IdModelState'
+    );
+    // guard
+    expect(symbols).toHaveProperty('2.type.properties.3.name', 'shouldOpen');
+    expect(symbols).toHaveProperty('2.type.properties.3.kind', 'property');
+    expect(symbols).toHaveProperty('2.type.properties.3.type.kind', 'function');
+    expect(symbols).toHaveProperty('2.type.properties.3.type.parameters.0.kind', 'parameter');
+    expect(symbols).toHaveProperty('2.type.properties.3.type.parameters.0.name', 'state');
+    expect(symbols).toHaveProperty('2.type.properties.3.type.parameters.0.type.kind', 'symbol');
+    expect(symbols).toHaveProperty(
+      '2.type.properties.3.type.parameters.0.type.name',
+      'IdModelState'
+    );
+
+    // model state
+    expect(symbols).toHaveProperty('3.name', 'IdModelState');
+
+    // model events
+    expect(symbols).toHaveProperty('4.name', 'IdModelEvents');
+  });
+
+  it('should replace instances of "ReturnType<typeof useMyModel>" to a symbol of MyModel', () => {
```

Comment:
I think this is inaccurate, maybe should be:
```suggestion
  it('should replace instances of "ReturnType<typeof useMyModel>" with a symbol of useMyModel', () => {
```